optimizely/optimizely_factory.py

```diff
@@ -0,0 +1,188 @@
+        """ Convenience method for setting the maximum time interval in milliseconds between event dispatches.
+        Args:
+          flush_interval: Time interval between event dispatches.
+          logger: Optional LoggerInterface Provides a log method to log messages.
+         """
+
+        if isinstance(flush_interval, bool):
+            logger.error('Flush interval is invalid, setting to default flush interval # {0}'.format(
+                BatchEventProcessor._DEFAULT_FLUSH_INTERVAL))
+            return
+
+        if not isinstance(flush_interval, int):
+            logger.error('Flush interval is invalid, setting to default flush interval # {0}'.format(
+                BatchEventProcessor._DEFAULT_FLUSH_INTERVAL))
+            return
+
+        if not flush_interval > 0:
+            logger.error('Flush interval is negative, setting to default flush interval # {0}'.format(
+                BatchEventProcessor._DEFAULT_FLUSH_INTERVAL))
+            return
+
+        OptimizelyFactory.max_event_flush_interval = flush_interval
+        return OptimizelyFactory.max_event_flush_interval
+
+    @staticmethod
+    def set_polling_interval(polling_interval):
+        """ Method to set frequency at which datafile has to be polled.
+            Args:
+              polling_interval: Time in seconds after which to update datafile.
+        """
+        OptimizelyFactory.polling_interval = polling_interval
+        return OptimizelyFactory.polling_interval
+
+    @staticmethod
+    def set_blocking_timeout(blocking_timeout):
+        """ Method to set time in seconds to block the config call until config has been initialized.
+            Args:
+              blocking_timeout: Time in seconds to block the config call.
+       """
+        OptimizelyFactory.blocking_timeout = blocking_timeout
+        return OptimizelyFactory.blocking_timeout
+
+    @staticmethod
+    def default_instance(sdk_key, datafile=None):
+        """ Returns a new optimizely instance..
+          Args:
+            sdk_key:  Required String uniquely identifying the fallback datafile corresponding to project.
+            datafile: Optional JSON string datafile.
+        """
+        error_handler = NoOpErrorHandler()
+        logger = optimizely_logger.NoOpLogger()
+        notification_center = NotificationCenter(logger)
+
+        config_manager_options = {
+            'sdk_key': sdk_key,
+            'update_interval': OptimizelyFactory.polling_interval,
+            'blocking_timeout': OptimizelyFactory.blocking_timeout,
+            'datafile': datafile,
+            'logger': logger,
+            'error_handler': error_handler,
+            'notification_center': notification_center,
+        }
+
+        config_manager = PollingConfigManager(**config_manager_options)
+
+        optimizely = Optimizely(
+            datafile, None, logger, error_handler, None, None, sdk_key, config_manager, notification_center
```

Comment:
Shouldn't we include "BatchEventProcessor" as a part of this factory method as a common configuration?